Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,393 @@ public class ClassificationTests: XCTestCase {
+      ]
+    )
+  }
+
+  public func testIfconfig2() {
+    assertClassification(
+      """
+      #if !CONF
+      #else
+      """,
+      expected: [
+        ClassificationSpec(source: "#if", kind: .poundDirective),
+        ClassificationSpec(source: "!", kind: .operatorIdentifier),
+        ClassificationSpec(source: "CONF", kind: .buildConfigId),
+        ClassificationSpec(source: "#else", kind: .poundDirective),
+      ]
+    )
+  }
+
+  public func testBuiltinMacros() {
+    assertClassification(
+      """
+      #error("Error")
+      #warning("Warning")
+      """,
+      expected: [
+        ClassificationSpec(source: "error", kind: .identifier),
+        ClassificationSpec(source: "\"Error\"", kind: .stringLiteral),
+        ClassificationSpec(source: "warning", kind: .identifier),
+        ClassificationSpec(source: "\"Warning\"", kind: .stringLiteral),
+      ]
+    )
+  }
+
+  public func testSourceLocation() {
+    assertClassification(
+      #"#sourceLocation(file: "x", line: 1)"#,
+      expected: [
+        ClassificationSpec(source: "#sourceLocation", kind: .poundDirective),
+        ClassificationSpec(source: "file", kind: .keyword),
+        ClassificationSpec(source: #""x""#, kind: .stringLiteral),
+        ClassificationSpec(source: "line", kind: .keyword),
+        ClassificationSpec(source: "1", kind: .integerLiteral),
+      ]
+    )
+  }
+
+  public func testColorLiteral() {
+    assertClassification(
+      "#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)",
+      expected: [
+        ClassificationSpec(source: "colorLiteral", kind: .identifier),
+        ClassificationSpec(source: "red", kind: .identifier),
+        ClassificationSpec(source: "0", kind: .integerLiteral),
+        ClassificationSpec(source: "green", kind: .identifier),
+        ClassificationSpec(source: "0", kind: .integerLiteral),
+        ClassificationSpec(source: "blue", kind: .identifier),
+        ClassificationSpec(source: "0", kind: .integerLiteral),
+        ClassificationSpec(source: "alpha", kind: .identifier),
+        ClassificationSpec(source: "1", kind: .integerLiteral),
+      ]
+    )
+  }
+
+  public func testImageLiteral() {
+    assertClassification(
+      "#imageLiteral(resourceName: \"cloud.png\")",
+      expected: [
+        ClassificationSpec(source: "imageLiteral", kind: .identifier),
+        ClassificationSpec(source: "resourceName", kind: .identifier),
+        ClassificationSpec(source: "\"cloud.png\"", kind: .stringLiteral),
+      ]
+    )
+  }
+
+  public func testFileLiteral() {
+    assertClassification(
+      "#fileLiteral(resourceName: \"cloud.png\")",
+      expected: [
+        ClassificationSpec(source: "fileLiteral", kind: .identifier),
+        ClassificationSpec(source: "resourceName", kind: .identifier),
+        ClassificationSpec(source: "\"cloud.png\"", kind: .stringLiteral),
+      ]
+    )
+  }
+
+  // TODO@hzy discuz about this and below
```

Comment:
LGTM. I would just prefer to also classify the `@` as `attribute`. AFAICT you just need revert one of the changes you did above.